File: 2022-2023/sourcecode/lab/lab13/DaggerHiltBasicsJava/app/src/main/java/pl/edu/uwr/pum/daggerhiltbasicsjava/data/remote/PlaceholderApi.java
package pl.edu.uwr.pum.daggerhiltbasicsjava.data.remote;

import java.util.List;

import pl.edu.uwr.pum.daggerhiltbasicsjava.data.Post;
import retrofit2.Call;
import retrofit2.http.GET;

public interface PlaceholderApi {

    @GET("posts")
    Call<List<Post>> getPosts();
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/daggerhiltbasicskotlin/data/remote/PlaceholderApi.kt
package pl.edu.uwr.pum.daggerhiltbasicskotlin.data.remote

import pl.edu.uwr.pum.daggerhiltbasicskotlin.data.Post
import retrofit2.Response
import retrofit2.http.GET

interface PlaceholderApi {
    @GET("posts")
    suspend fun posts(): Response<List<Post>>
}